sdk/src/transaction/versioned.rs

```diff
@@ -0,0 +1,76 @@
+//! Defines a transaction which supports multiple versions of messages.
+
+#![cfg(feature = "full")]
+
+use {
+    crate::{
+        hash::Hash,
+        message::VersionedMessage,
+        sanitize::{Sanitize, SanitizeError},
+        short_vec,
+        signature::Signature,
+        transaction::{Result, Transaction, TransactionError},
+    },
+    serde::Serialize,
+};
+
+// NOTE: Serialization-related changes must be paired with the direct read at sigverify.
+/// An atomic transaction
+#[derive(Debug, PartialEq, Default, Eq, Clone, Serialize, Deserialize, AbiExample)]
+pub struct VersionedTransaction {
+    /// List of signatures
+    #[serde(with = "short_vec")]
+    pub signatures: Vec<Signature>,
+    /// Message to sign.
+    pub message: VersionedMessage,
+}
+
+impl Sanitize for VersionedTransaction {
+    fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
+        if self.message.header().num_required_signatures as usize > self.signatures.len() {
+            return Err(SanitizeError::IndexOutOfBounds);
+        }
+        if self.signatures.len() > self.message.unmapped_keys_len() {
+            return Err(SanitizeError::IndexOutOfBounds);
+        }
+        self.message.sanitize()
```

Comment:
Should we be referencing `self.message` members before they've been sanitized?

Reply:
In this case it's not consequential but I updated the order anyways